packages/x-data-grid/src/constants/cssVariables.ts

```diff
@@ -0,0 +1,168 @@
+// NOTE: Breakpoints can't come from the theme because we need access to them at
+// initialization time and media-queries can't use CSS variables. For users with
+// custom breakpoints, we might want to provide a way to configure them globally
+// instead of through the theme.
+const breakpoints = {
+  values: {
+    xs: 0, // phone
+    sm: 600, // tablet
+    md: 900, // small laptop
+    lg: 1200, // desktop
+    xl: 1536, // large screen
+  },
+  up: (key: any) => {
+    const values = breakpoints.values as any;
+    const value = typeof values[key] === 'number' ? values[key] : key;
+    return `@media (min-width:${value}px)`;
+  },
+};
+
+const keys = {
+  spacingUnit: '--DataGrid-t-spacing-unit',
+
+  /* Variables */
+  colors: {
+    border: {
+      base: '--DataGrid-t-colors-border-base',
+    },
+    foreground: {
```

Comment:
How about calling it `text` to make it clearer for the average user?

Reply:
This is not meant to be used by the average user, only by design-system maintainers. And I like the symmetry between background/foreground.